### PR TITLE
Spin up and tear down testing EC2 instances

### PR DESCRIPTION
**Pull request by jsubirat**
This PR creates and destroys the test executor VMs during the "Run E2E tests" workflow.
- All the VMs have been (manually) tested to be SSM-accessible once created (this will be properly tested in the next PR, where we'll be managing them with Ansible).
- CentOS 8 and 9 images use CentOS Stream, while CentOS 7 keeps using CentOS.
- All the AMIs have been adapted for the ones in the `us-east-2` region.

### CODE REVIEW

File: terraform/ec2-instances-creator/user_data_script_for_ssm.tftpl [hunk 1]
```diff
@@ -0,0 +1,15 @@
+#!/bin/bash
+
+%{if os_distro == "centos"}
+sudo %{if os_version == 7}yum%{else}dnf%{endif} install -y https://s3.amazonaws.com/ec2-downloads-windows/SSMAgent/latest/linux_%{if arch == "x86_64"}amd64%{else}arm64%{endif}/amazon-ssm-agent.rpm
```
Comment (jsubirat) [inline terraform/ec2-instances-creator/user_data_script_for_ssm.tftpl:4]: CentOS 7 still use "yum", while newer versions switched to using "dnf"

File: terraform/ec2-test-executors/main.tf [hunk 2]
```diff
@@ -0,0 +1,11 @@
+locals {
+  all_instances_matrix_path = "../../versions/strategyMatrix.json"
```
Comment (jsubirat) [inline terraform/ec2-test-executors/main.tf:2]: We spin up all instances in one shot, but we'll be potentially provisioning them differently using Ansible later.

File: terraform/ec2-test-executors/terraform.backend.tf.dist [hunk 3]
```diff
@@ -0,0 +1,11 @@
+#########################################
+# State Backend
+#########################################
+terraform {
+  backend "s3" {
+    encrypt = true
+    bucket  = "logging-e2e-testing-tf"
+    key     = "test-executors-pr-PR_NUMBER"
```
Comment (jsubirat) [inline terraform/ec2-test-executors/terraform.backend.tf.dist:8]: This uses a differentiated state key with regards to the SUSE builder VMs.

File: versions/amazonlinux_2023.yml [hunk 4]
```diff
@@ -0,0 +1,9 @@
+osDistro: amazonlinux
+osVersion: 2023
+# First available AL2023 version with both x86_64 and aarch64 support is 2.1.5
+fbVersion: 2.1.9
```
Comment (jsubirat) [inline versions/amazonlinux_2023.yml:4]: Note I had to do this override for Amazon Linux 2023 only

**Review by luckslovez — APPROVED** [review]
LGTM! 👏🏽

**Pull request merged**